### DISCUSSION TITLE
AJV is not allowed in Edge Runtime

### DISCUSSION THREAD

**Discussion by anarkrypto** [vercel/next.js#47063] [Help]
### Summary

I can't use AJV with edge runtime. This is the error:

```bash
Dynamic Code Evaluation (e. g. 'eval', 'new Function', 'WebAssembly.compile') not allowed in Edge Runtime
```

![image](https://user-images.githubusercontent.com/32111208/224667006-ce43419e-0b7f-415d-a376-62f8f97e937b.png)

Is it possible to get around it?

### Additional information

```js
next: 13.2.3
ajv: 8.12.0
```

**Comment by anarkrypto** [comment]
https://ajv.js.org/standalone.html

This article appears to provide a solution.

It seems to solve it, but it clearly adds substantial complexity. It may make it unfeasible to use AJV without a more practical solution, such as auto pre-compiling during a pre-build of next.

**Comment by anarkrypto** [comment]
Another solution I found, cfworker json-schema: 
https://github.com/cfworker/cfworker/tree/main/packages/json-schema

A AJV like JSON schema validator that run on Cloudflare workers.

A major disadvantage, however, is the error outputs. Does not have functionality like ajv.errorsText()

So while AJV can produce an error message like:
```
data/name must be string
```
cfworker produce: 

```
[
    {
      instanceLocation: '#',
      keyword: 'properties',
      keywordLocation: '#/properties',
      error: 'Property "name" does not match schema.'
    },
    {
      instanceLocation: '#/name',
      keyword: 'type',
      keywordLocation: '#/properties/name/type',
      error: 'Instance type "number" is invalid. Expected "string".'
    },
    {
      instanceLocation: '#',
      keyword: 'additionalProperties',
      keywordLocation: '#/additionalProperties',
      error: 'Property "name" does not match additional properties schema.'
    },
    {
      instanceLocation: '#/name',
      keyword: 'false',
      keywordLocation: '#/name',
      error: 'False boolean schema.'
    }
  ]
```

**Comment by icyJoseph** [comment]
There's one way, but it doesn't fully `un-node` just yet: https://github.com/unjs/unenv

Although your issue has to do with dynamic code evaluation, mmmm
